src/impy/kinematics.py

```diff
@@ -185,57 +100,60 @@ def __init__(
         if particle1 is None or particle2 is None:
             raise ValueError("particle1 and particle2 must be set")
 
-        self.p1 = _normalize_particle(particle1)
-        self.p2 = _normalize_particle(particle2)
+        self.p1 = process_particle(particle1)
+        self.p2 = process_particle(particle2)
 
         if isinstance(self.p1, CompositeTarget):
             raise ValueError("Only 2nd particle can be CompositeTarget")
 
         p2_is_composite = isinstance(self.p2, CompositeTarget)
 
-        m1 = mass(self.p1)
+        m1 = nucleon_mass if self.p1.is_nucleus else mass(self.p1)
         m2 = nucleon_mass if p2_is_composite or self.p2.is_nucleus else mass(self.p2)
 
         self.beams = (np.zeros(4), np.zeros(4))
 
-        # Input specification in center of mass frame
+        # Input specification in center-of-mass frame
         if ecm is not None:
-            self.frame = EventFrame.CENTER_OF_MASS if frame is None else frame
+            self.frame = frame or EventFrame.CENTER_OF_MASS
             self.ecm = ecm
-            self.elab = 0.5 * (ecm**2 - m1**2 - m2**2) / m2
```

Comment:
@afedynitch  You asked about typos. See lines 205 and 219
205 - m1 ** 2 - m2 ** 2
219 - m1 ** 2 + m2 ** 2
I believe the second is correct. I recalculated this and put the computation in the comment in impy.util.

Reply:
Uhh. ok. Thanks.